### ISSUE TITLE
sed command in RestartDnsmasq for redhat hosts does insert 127.0.0.1 before each nameserver entry

### ISSUE THREAD

**Issue by hferentschik** [vagrant-landrush/landrush#234]

**Comment by hferentschik** [comment]
The problem is the used sed expression:

          SED_COMMAND = <<-EOF.gsub(/^ +/, '')
          sudo sed -i.orig '/nameserver/i\\
          nameserver 127.0.0.1 # Added by landrush, a vagrant plugin

          ' /etc/resolv.conf
          EOF

It does a global search and replace. The line should only be inserted once at the top. Something like this should work and be simpler:

     SED_COMMAND = "sudo sed -i.orig '1 i\\nameserver 127.0.0.1 # Added by landrush, a vagrant plugin' /etc/resolv.conf"

**Comment by praveenkumar** [comment]
> Something like this should work and be simpler:

That expression gave below error

```
sed: -e expression #1, char 56: extra characters after command
```

What works is below expression. (Also tested.)

```
SED_COMMAND = <<-EOF.gsub(/^ +/, '') 
sudo sed -i.orig '1 i\
nameserver 127.0.0.1 # Added by landrush, a vagrant plugin
' /etc/resolv.conf
```

**Comment by hferentschik** [comment]
@praveenkumar Can you try:

          SED_COMMAND = <<-EOF.gsub(/^ +/, '')
          sudo sed -i.orig '1 i\
          # Added by landrush, a vagrant plugin \
          nameserver 127.0.0.1 \
          ' /etc/resolv.conf
          EOF

**Comment by praveenkumar** [comment]
> @praveenkumar Can you try:

This will give below result

```
# Added by landrush, a vagrant plugin           nameserver 127.0.0.1
```

**Comment by hferentschik** [comment]
Needs to be \\

Can you try:

          SED_COMMAND = <<-EOF.gsub(/^ +/, '')
          sudo sed -i.orig '1 i\
          # Added by landrush, a vagrant plugin \\
          nameserver 127.0.0.1 \\
          ' /etc/resolv.conf
          EOF

**Comment by hferentschik** [comment]
fixed by pull request #235
